File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.py
def solution(new_id):
    new_id=new_id.lower()
    ids = ''
    for id in new_id:
        if id.isalpha() or id.isdigit() or id in "-_.":
            ids+=id

    while '..' in ids:
        ids = ids.replace('..','.')

    if ids and ids[0] == '.':
        ids = ids[1:]
    if ids and ids[-1] =='.':
        ids = ids[:-1]
    if ids == '':
        ids ='a'
    ids = ids[:15]
    if ids[-1] =='.':
        ids = ids[:-1]
    while len(ids)<3:
        ids+=ids[-1]
    return ids

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.java
import java.util.*;
class Solution {
    public String solution(String new_id) {
        String answer = "";
        String id = new_id.toLowerCase();

        StringBuilder sb = new StringBuilder();
        int cnt = 0;
        char prev = ' ';
        for (int i = 0; i < id.length(); i++) {
            char c = id.charAt(i);
            if (Character.isLetter(c) || Character.isDigit(c) || c == '-' || c == '_' || c == '.') {
                if (c == '.' && prev == '.') {
                    continue; // 연속된 .을 방지
                }
                sb.append(c);
                prev = c; // 현재 문자를 prev로 저장
            }
        }
        
        while(sb.length() > 0 && sb.charAt(0)=='.'){
            sb.deleteCharAt(0);
        }

        while(sb.length() > 0&&sb.charAt(sb.length()-1)=='.'){
            sb.deleteCharAt(sb.length() - 1);
        }
        if(sb.length()==0){
            sb.append('a');
        }
        if(sb.length()>=16){
            sb.setLength(15);
        }
        while(sb.length() > 0&&sb.charAt(sb.length()-1)=='.'){
            sb.deleteCharAt(sb.length() - 1);
        }
        if(sb.length()<=2){
            char last = sb.charAt(sb.length()-1);
            while(sb.length() < 3){
                sb.append(last);
            }
        }
        System.out.println(sb);
        
        answer = sb.toString();
        return answer;
    }
}